lib/jbuilder/collection_renderer.rb

```diff
@@ -0,0 +1,118 @@
+require 'delegate'
+require 'active_support/concern'
+
+begin
+  require 'action_view/renderer/collection_renderer'
+rescue LoadError
+  require 'action_view/renderer/partial_renderer'
+end
+
+class Jbuilder
+  module CollectionRenderable # :nodoc:
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def supports_collection_rendering?
+        superclass.private_method_defined?(:build_rendered_template) && superclass.private_method_defined?(:build_rendered_collection)
+      end
+    end
+
+    private
+
+    def build_rendered_template(content, template, layout = nil)
+      super(content || json.attributes!, template)
+    end
+
+    def build_rendered_collection(templates, _spacer)
+      json.merge!(templates.map(&:body))
+    end
+
+    def json
+      @options[:locals].fetch(:json)
+    end
+
+    class ScopedIterator < ::SimpleDelegator # :nodoc:
+      include Enumerable
+
+      def initialize(obj, scope)
+        super(obj)
+        @scope = scope
+      end
+
+      # Rails 6.0 support:
+      def each
+        return enum_for(:each) unless block_given?
+
+        __getobj__.each do |object|
+          @scope.call { yield(object) }
+        end
+      end
+
+      # Rails 6.1 support:
+      def each_with_info
+        return enum_for(:each_with_info) unless block_given?
+
+        __getobj__.each_with_info do |object, (path, as, counter, iteration)|
+          @scope.call { yield(object, [path, as, counter, iteration]) }
+        end
+      end
+    end
+
+    private_constant :ScopedIterator
+  end
+
+  collection_renderer_available = begin
+                                     ::ActionView::CollectionRenderer
+
+                                     true
+                                   rescue ::NameError
+                                     false
+                                   end
+
+  if collection_renderer_available
```

Comment:
Could we replace all this with?

```suggestion
  if defined?(::ActionView::CollectionRenderer)
```

If not, our style is:

```suggestion
  collection_renderer_available =
    begin
      ::ActionView::CollectionRenderer
      true
    rescue ::NameError
      false
    end

  if collection_renderer_available
```